uefi-hello-world-aarch64/build.py

```diff
@@ -0,0 +1,405 @@
+        return ovmf_dir
+
+    if sys.platform.startswith('linux'):
+        possible_paths = [
+            # Most distros, including CentOS, Fedora, Debian, and Ubuntu.
+            Path('/usr/share/OVMF'),
+            # Arch Linux
+            Path('/usr/share/ovmf/x64'),
+        ]
+        for path in possible_paths:
+            if check_ovmf_dir(path):
+                return path
+
+    raise FileNotFoundError(f'OVMF files not found anywhere')
+
+
+def run_qemu():
+    'Runs the code in QEMU.'
+
+    # Rebuild all the changes.
+    build('--features', 'qemu')
+
+    ovmf_code, ovmf_vars = ovmf_files(find_ovmf())
+
+    examples_dir = build_dir() / 'examples'
+
+    qemu_monitor_pipe = 'qemu-monitor'
+
+    qemu_flags = [
+        # Disable default devices.
+        # QEMU by defaults enables a ton of devices which slow down boot.
+        # '-nodefaults',
+
+        # Use a generic ARM environment. Sadly qemu can't emulate a RPi 4 like machine though
+        '-machine', 'virt',
+
+        # a57 is a very generic 64-bit ARM CPU in the wild
+        '-cpu', 'cortex-a72',
+
+        # Multi-processor services protocol test needs exactly 3 CPUs.
+        '-smp', '4',
+
+        # Allocate some memory.
+        '-m', '512M',
+
+        # Set up OVMF.
+        '-drive', f'if=pflash,format=raw,file={ovmf_code},readonly=on',
+        '-drive', f'if=pflash,format=raw,file={ovmf_vars},readonly=on',
+
+        # Mount a local directory as a FAT partition.
+        '-drive', f'format=raw,file=fat:rw:{esp_dir()}',
+
+        # Mount the built examples directory.
+        '-drive', f'format=raw,file=fat:rw:{examples_dir}',
+    ]
+
+    if sys.platform.startswith('win'):
+        qemu_flags = qemu_flags + [
+            # there's currently no way to get debug information from qemu-system-aarch64
+            # so I commented them out just as of right now. I'm running it off Windows 10 however,
+            # if you happen to know some tricks please consider giving a helping hand
+
+            # setup a character device char0 to print a combined result of stdio
+            '-chardev', 'stdio,mux=on,id=char0',
+
+            # set the monitor to use character device char0
+            '-mon', 'chardev=char0,mode=readline',
+        ]
+    if sys.platform.startswith('linux'):
+        qemu_flags = qemu_flags + [
+            # Connect the serial port to the host. OVMF is kind enough to connect
+            # the UEFI stdout and stdin to that port too.
+            '-serial', 'stdio',
+
+            # Map the QEMU exit signal to port f4
+            '-device', 'isa-debug-exit,iobase=0xf4,iosize=0x04',
+
+            # Map the QEMU monitor to a pair of named pipes
+            '-qmp', f'pipe:{qemu_monitor_pipe}',
+
+            # OVMF debug builds can output information to a serial `debugcon`.
+            # Only enable when debugging UEFI boot:
+            '-debugcon', 'file:debug.log',
+
+            '-global', 'isa-debugcon.iobase=0x402',
```

Comment:
I think these should be commented, since QEMU doesn't support the isa-debug-port/connection on ARM.